src/DynamoDbModel.php

```diff
@@ -155,6 +155,37 @@ public function save(array $options = [])
         return $saved;
     }
 
+    public function saveAsync(array $options = [])
+    {
+        $create = !$this->exists;
+
+        if ($this->fireModelEvent('saving') === false) {
+            return false;
+        }
+
+        if ($create && $this->fireModelEvent('creating')  === false) {
+            return false;
+        }
+
+        if (!$create && $this->fireModelEvent('updating') === false) {
+            return false;
+        }
+
+        if ($this->usesTimestamps()) {
+            $this->updateTimestamps();
+        }
+
+        $savePromise = $this->newQuery()->saveAsync();
+
+        $this->exists = true;
```

Comment:
Don't think we can execute any thing from here onwards since the promise is not resolved yet. Can we wrap the promise to perform these steps after it is resolved?

i.e in JS version, it'd be:

```js
const promise = this.newQuery().saveAsync();

return promise
  .then(() => {
    this.exists = true;
    this.wasRecentlyCreated = create;
    this.fireModelEvent(...);
    this.finishSave(options);
    return true;
  })
  .catch(() => false);
```